### DISCUSSION TITLE
Empty reports for Userinfo-exhaust

### DISCUSSION THREAD

**Discussion by raghupathiguduri** [Sunbird-Obsrv/Community#30] [Installation]
Hi Team, 
We have installed the sunbird release-4.7.0 with all data-analytics services (analytics,spark,flinksjobs,druid,secor). We have enabled the exhaust reports. We found that progress-exhaust and response-exhaust reports are working fine but there is an issue with user-info-exhaust report.
Report execution on spark is generating the empty reports for user-info-exhaust. We have users shared the consent using consent-popup in the course.
How to verify the data in Databases related to this reports ?  What data is should have in Redis & Cassandra to generate this report ? How the data gets ingested into the required databases ? Any prerequisites for this report from organisation settings ?

**Comment by SanthoshVasabhaktula** [comment]
@raghupathiguduri 

The user cache should be populated in the redis. In ideal scenario if the data pipeline is setup and the users are created the cache would have been populated. Can you check if the cache is populated or not?

If not, you can run a script to populate the cache one time and ensure the pipeline is running to populate the cache going forward.

@manjudr - Can you share the details about the script?

**Comment by manjudr** [comment]
@raghupathiguduri  - We have an ETL Job that can help populate the user data into the Redis and also we have the Jenkins job. 
Populate the user details into the cache and then run the exhaust job.
Here are the Details. 
Script link - https://github.com/Sunbird-Ed/sunbird-data-products/tree/release-5.0.0/etl-jobs/src
Jenkins Job Name - ETLUserCacheIndexer. 
Ansible Role - https://github.com/Sunbird-Obsrv/sunbird-data-pipeline/tree/release-5.0.0/ansible/roles/content-snapshot-indexer
